app/status.go

```diff
@@ -236,10 +236,21 @@ func IsUserAway(lastActivityAt int64) bool {
 	return model.GetMillis()-lastActivityAt >= *utils.Cfg.TeamSettings.UserStatusAwayTimeout*1000
 }
 
-func DoesStatusAllowPushNotification(user *model.User, status *model.Status, channelId string) bool {
+func DoesStatusAllowPushNotification(user *model.User, channelNotifyProps model.StringMap, wasMentioned bool, status *model.Status, channelId string) bool {
 	props := user.NotifyProps
 
-	if props["push"] == "none" {
+	channelNotify := ""
+	ok := false
+	if channelNotify, ok = channelNotifyProps[model.PUSH_NOTIFY_PROP]; ok {
+		if channelNotify == model.USER_NOTIFY_NONE {
+			return false
+		} else if channelNotify == model.CHANNEL_NOTIFY_MENTION && !wasMentioned {
+			return false
+		}
+	}
+
+	if props[model.PUSH_NOTIFY_PROP] == model.USER_NOTIFY_NONE &&
```

Comment:
Shouldn't there be wasMentioned logic like above? So this currently would not catch the case where the default is CHANNEL_NOTIFY_MENTION and the person was not mentioned.